src/coreclr/src/jit/hwintrinsic.cpp

```diff
@@ -643,37 +639,85 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
         assert(sizeBytes != 0);
     }
 
-    // NOTE: The following code assumes that for all intrinsics
-    // taking an immediate operand, that operand will be last.
-    if (sig->numArgs > 0 && HWIntrinsicInfo::isImmOp(intrinsic, impStackTop().val))
+    GenTree* immOp = nullptr;
+
+#ifdef TARGET_ARM64
+    if (intrinsic == NI_AdvSimd_Insert)
```

Comment:
This should likely be tweaked a bit more, ARM64 will have cases with 2 immediate operands at varying positions.

Reply:
Yes, I realized this yesterday - for a case `ins Vd.T[imm1], Vs.T[imm2]` - perhaps this could wait until I start working on that intrinsic so I could get `Insert` and `Extract` in - Kunal is blocked on these.